src/UNL/UCBCN/Manager/BulkAction.php

```diff
@@ -0,0 +1,61 @@
+<?php
+namespace UNL\UCBCN\Manager;
+
+use UNL\UCBCN\Calendar;
+use UNL\UCBCN\Calendar\Event as CalendarHasEvent;
+use UNL\UCBCN\Event;
+
+class BulkAction implements PostHandlerInterface
+{
+    public $options = array();
+    public $calendar;
+
+    public function __construct($options = array()) 
+    {
+        $this->options = $options + $this->options;
+        $this->calendar = Calendar::getByShortName($this->options['calendar_shortname']);
+        if ($this->calendar === FALSE) {
+            throw new \Exception("That calendar could not be found.", 404);
+        }
+    }
+
+    public function handlePost(array $get, array $post, array $files)
+    {
+        // get the ids of the events to do things with
+        $ids = explode(',', $post['ids']);
+        $action = $post['action'];
```

Comment:
I think you are still planning to do this, but checking permissions would be good here too :)